provers/evm-prover/src/main.rs

```diff
@@ -0,0 +1,191 @@
+// Import the generated proto rust code
+pub mod prover {
+    tonic::include_proto!("celestia.prover.v1");
+}
+
+use prover::prover_server::{Prover, ProverServer};
+use prover::{
+    ProveMembershipRequest, ProveMembershipResponse, ProveStateTransitionRequest,
+    ProveStateTransitionResponse,
+};
+
+use celestia_rpc::{BlobClient, Client, HeaderClient};
+use celestia_types::{nmt::Namespace, Blob};
+use rsp_client_executor::io::ClientExecutorInput;
+use sp1_sdk::{include_elf, ProverClient, SP1Stdin};
+
+use ethers::{
+    providers::{Http, Middleware, Provider},
+    types::BlockNumber,
+};
+
+/// The ELF file for the Succinct RISC-V zkVM.
+const BLEVM_ELF: &[u8] = include_elf!("blevm");
+
+pub struct ProverService {
+    celestia_client: Client,
+    evm_client: Provider<Http>,
+    sp1_prover: ProverClient,
+}
+
+impl ProverService {
+    async fn new() -> Result<Self, Box<dyn std::error::Error>> {
+        let token = env::var("CELESTIA_NODE_AUTH_TOKEN").expect("Token not provided");
+        let celestia_client = Client::new("ws://localhost:26658", Some(&token))
+            .await
+            .expect("Failed creating Celestia RPC client");
+
+        let evm_rpc = env::var("EVM_RPC_URL").expect("EVM_RPC_URL not provided");
+        let evm_client = Provider::try_from(evm_rpc)?;
+
+        Ok(ProverService {
+            celestia_client,
+            evm_client,
+            sp1_prover: ProverClient::new(),
+        })
+    }
+
+    async fn get_latest_block_number(&self) -> Result<u64, Status> {
+        self.evm_client
+            .get_block(BlockNumber::Latest)
+            .await
+            .map_err(|e| Status::internal(format!("Failed to get latest block: {}", e)))?
+            .ok_or_else(|| Status::internal("No latest block found"))?
+            .number
+            .ok_or_else(|| Status::internal("Block has no number"))?
+            .as_u64()
+            .try_into()
+            .map_err(|e| Status::internal(format!("Failed to convert block number: {}", e)))
+    }
+}
+
+#[tonic::async_trait]
+impl Prover for ProverService {
+    async fn prove_state_transition(
+        &self,
+        request: Request<ProveStateTransitionRequest>,
+    ) -> Result<Response<ProveStateTransitionResponse>, Status> {
+        println!("Got state transition request: {:?}", request);
+        // Get the latest block number from EVM rollup
+        let latest_height = self.get_latest_block_number().await?;
+
+        // Load the zkEVM input from the file
+        let input_bytes = fs::read(format!("input/1/{}.bin", latest_height))
+            .map_err(|e| Status::internal(format!("Failed to read input file: {}", e)))?;
+
+        let input: ClientExecutorInput = bincode::deserialize(&input_bytes)
+            .map_err(|e| Status::internal(format!("Failed to deserialize input: {}", e)))?;
+
+        // Use the namespace from the request or a default
+        let namespace = Namespace::new_v0(&alloy::hex::decode("0f0f0f0f0f0f0f0f0f0f").unwrap())
```

Comment:
nit: this doesn't use the namespace from the request yet, right?

```suggestion
        // Use a default namespace
        // TODO: use the namespace from the request
        let namespace = Namespace::new_v0(&alloy::hex::decode("0f0f0f0f0f0f0f0f0f0f").unwrap())
```

Reply:
We're parsing this from env now.